builder/builder.go

```diff
@@ -158,19 +159,37 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 	bf := func(childCtx context.Context, gwClient gwclient.Client) (*gwclient.Result, error) {
 		var err error
 		if !b.builtMain {
-			mts, err = earthfile2llb.Earthfile2LLB(childCtx, target, earthfile2llb.ConvertOpt{
-				GwClient:             gwClient,
-				Resolver:             b.resolver,
-				ImageResolveMode:     b.opt.ImageResolveMode,
-				DockerBuilderFun:     b.MakeImageAsTarBuilderFun(),
-				CleanCollection:      b.opt.CleanCollection,
-				Platform:             opt.Platform,
-				VarCollection:        b.opt.VarCollection,
-				BuildContextProvider: b.opt.BuildContextProvider,
-				CacheImports:         b.opt.CacheImports,
-				UseInlineCache:       b.opt.UseInlineCache,
-				UseFakeDep:           b.opt.UseFakeDep,
-			})
+			if b.opt.EnableAst {
```

Comment:
It took me a while to see they only difference between the if and the else is just Earthfile2LLB2 vs Earthfile2LLB; where as the convert options are identical. Do we expect that the ConvertOpt values will ever change between the two different calls? If not is is possible to move them outside of the if/else

Reply:
Yeah, I should do that - I left this a bit messy.

Will fix.

Reply:
Fixed now - I merged the two earthfile2llb's to clean things up.